### PR TITLE
 Styling for Student Strategies sidebar 

### PR DESCRIPTION
**Pull request by Zmuse001**
### Changes
styled the student strategies sidebar to look like the screenshot below and compiled the tailwind

### Link to Issue 
https://app.asana.com/0/1206510424537042/1206645498622461/f

### Comments, Notes, or  Questions
running the compiler made the page look different? it all squished now.

### Related Screenshots 
Screenshot 2024-05-08 at 3 31 20 PM

### Before you submit for review, Did you...

- [ ] Added relevant tests (if applicable)
- [ ] Linked Asana ticket to PR
- [ ] Added comments, notes, or questions for reviewwer
- [ ] Added relevant screenshots
- [ ] Ensured ticket title is clear and concise(i.e. 'Donation Form Feature' instead of 'sfg/chels/donation-form'

### CODE REVIEW

**Review by KNAthari — DISMISSED** [review]
It looks good.

**Comment by camckin10** [comment]
@Zmuse001  You left a comment about running the compiler making the page look weird? 
Can you please include a screenshot of  what you see on your end locally when you run the compiler?

**Comment by Zmuse001** [comment]
> @Zmuse001 You left a comment about running the compiler making the page look weird? Can you please include a screenshot of what you see on your end locally when you run the compiler?

Screenshot 2024-05-09 at 10 55 31 AM

File: app/views/layouts/_sidebar_nav.html.erb [hunk 1]
```diff
@@ -12,6 +12,10 @@
           <%= link_to students_path, class: "nav-tab" do %>
             <!--add my_students icon here-->
             <img src="<%= asset_path("new_sidebar_icons/my_students_icon.png") %>" alt="my_students_icon" class="h-24 px-5"/>
+<<<<<<< HEAD
```
Comment (camckin10) [inline app/views/layouts/_sidebar_nav.html.erb:15]: This needs to get removed; This is from merging conflicts on this file.

File: app/views/styleguides/tempCodeRunnerFile.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="12" viewBox="0 0 16 12" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (camckin10) [inline app/views/styleguides/tempCodeRunnerFile.erb:1]: Whats the significance of adding this code?

File: app/views/students/_strategies.html.erb [hunk 3]
```diff
@@ -1,3 +1,6 @@
+
```
Comment (camckin10) [inline app/views/students/_strategies.html.erb:1]: Remove this extra whitespace

File: app/views/students/show.html.erb [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
     <%= image_tag("Emmett-no 2.png", alt: "Emmett Image", class: "h-1/2 w-2/5 mr-20 mt-40") %>
 
-    <div class="py-4 px-2 rounded-lg w-96 h-5/6 border border-gray-500 flex flex-col justify-between">
+    <div class="py-4 px-2 rounded-lg w-2/5 h-5/6 border border-gray-500 flex flex-col justify-between">
```
Comment (camckin10) [inline app/views/students/show.html.erb:11]: Did you change the width in this sidebar to fit all of the content you added?

File: app/views/students/_know.html.erb [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
-</button>
+<div class="w-fit">
+  <hr>
+  <div class="w-max flex justify-between items-center mb-2">
+    <%= image_tag 'basic_strategy_icons/head-side-gear-duotone 1.png', alt: 'Gear in head icon', class: 'pill', style: 'max-width: 45px; height: auto;' %>
+    <p class="text-white ml-2 text-sm">Ways of Being</p>
+    <button class="strategy-btn bg-gray-300 border-2 border-gray-600 hover:bg-gray-400 text-black font-bold px-2 rounded text-sm">
+      Know
+    </button>
+    <button class="strategy-btn bg-accent-blue-dark border-2 border-gray-600 hover:bg-green-600 text-white font-bold px-2 rounded flex justify-end items-center space-x-1 text-sm">
```
Comment (camckin10) [inline app/views/students/_know.html.erb:9]: The 'strategy-btn' class is not being used in this file. You can remove it, or rename each button by its part of the strategy (i.e. 'know-btn', 'info-btn', etc.) 
```suggestion
    <button class=" bg-accent-blue-dark border-2 border-gray-600 hover:bg-green-600 text-white font-bold px-2 rounded flex justify-end items-center space-x-1 text-sm">
```

File: app/views/students/_know.html.erb [hunk 6]
```diff
@@ -1,3 +1,47 @@
-<button class="bg-gray-300 border-2 border-gray-600 hover:bg-gray-400 text-black font-bold px-2 rounded">
-  Know
-</button>
+<div class="w-fit">
```
Comment (camckin10) [inline app/views/students/_know.html.erb:1]: @Zmuse001  This is a good first pass!  
The main issue seems to be the spacing in the sidebar. 
To clean this up more, you can wrap the content inside the sidebar in a grid. Adding a grid ensures that there are equal width/height rows, and columns. 
I spun up a sample TailwindPlay for what I was picturing:[example]( https://play.tailwindcss.com/Lypg7rH41B)
After you make these adjustments, then test the width and spacing of the content inside the sidebar to see if that resolves the spacing.   

A few more edits: 
--you don't need the use of 'justify-end' on the buttons if you place the content in the grid. 
--the 'know' is styled like a button, but needs to be styled like a pill

File: app/views/styleguides/tempCodeRunnerFile.erb (hunk 2 above)
Comment (Zmuse001) [reply app/views/styleguides/tempCodeRunnerFile.erb:1]: this is the code for the arrows

File: app/views/students/show.html.erb (hunk 4 above)
Comment (Zmuse001) [reply app/views/students/show.html.erb:11]: yes, i made it bigger to contain all the info

File: app/views/students/_know.html.erb (hunk 6 above)
Comment (Zmuse001) [reply app/views/students/_know.html.erb:1]: do i put the first div that contains it all in a grid or each individual div?

File: app/views/styleguides/tempCodeRunnerFile.erb (hunk 2 above)
Comment (camckin10) [reply app/views/styleguides/tempCodeRunnerFile.erb:1]: @Zmuse001 Ok! Can you please rename the file to reflect that it's the code for the arrows?

File: app/views/students/_know.html.erb (hunk 6 above)
Comment (camckin10) [reply app/views/students/_know.html.erb:1]: You would make the first div, on line #1  the div that would have the grid. 
The remaining divs would have class names beginning with 'col.'
Refer back to the Tailwindplay I initially posted.

File: app/views/styleguides/tempCodeRunnerFile.erb (hunk 2 above)
Comment (Zmuse001) [reply app/views/styleguides/tempCodeRunnerFile.erb:1]: ooh actually sorry that code has nothing to do with me. i don't remember touching that but i'll get rid of it.

File: app/views/layouts/_sidebar_nav.html.erb [hunk 7]
```diff
@@ -12,6 +12,7 @@
           <%= link_to students_path, class: "nav-tab" do %>
             <!--add my_students icon here-->
             <img src="<%= asset_path("new_sidebar_icons/my_students_icon.png") %>" alt="my_students_icon" class="h-24 px-5"/>
+            <i class="fa-brands fa-font-awesome"></i>
```
Comment (edithemmings) [inline app/views/layouts/_sidebar_nav.html.erb:15]: I don't think this should be added as a part of your ticket

File: app/views/students/_know.html.erb [hunk 8]
```diff
@@ -1,3 +1,50 @@
...
+    <%= image_tag 'basic_strategy_icons/head-side-gear-duotone 1.png', alt: 'Gear in head icon', class: 'pill', style: 'max-width: 45px; height: auto;' %>
+    <p class="text-white ml-2 text-sm">Ways of Being</p>
+    <button class="strategy-btn bg-gray-300 border-2 border-gray-600 hover:bg-gray-400 text-black font-bold px-2 rounded-full text-sm">
+      Know
+    </button>
+<<<<<<< HEAD
+    <button class=" bg-accent-blue-dark border-2 border-gray-600 hover:bg-green-600 text-white font-bold px-2 rounded flex justify-end items-center space-x-1 text-sm">
+=======
+    <button class="strategy-btn bg-accent-blue-dark border-2 border-gray-600 hover:bg-green-600 text-white font-bold px-2 rounded flex items-center space-x-1 text-sm">
+>>>>>>> c3cb04e (fixed errors in show.html and sidebar partial)
```
Comment (edithemmings) [inline app/views/students/_know.html.erb:13]: ^^ something goofy happened with the merge

**Comment by camckin10** [comment]
I am getting two different versions of your sidebar on my end: 
Screen Shot 2024-05-14 at 2 02 09 PM

@Zmuse001  What do the changes look like on your local site?

**Comment by KNAthari** [comment]
> I am getting two different versions of your sidebar on my end: Screen Shot 2024-05-14 at 2 02 09 PM
> 
> @Zmuse001 What do the changes look like on your local site?

That is odd, because I am not getting that on my local server.

**Review by KNAthari — COMMENTED** [review]
Okay, now, I am having some spacing issues on my end.
Screenshot 2024-05-14 at 2 27 26 PM

**Comment by Zmuse001** [comment]
> Okay, now, I am having some spacing issues on my end. Screenshot 2024-05-14 at 2 27 26 PM

this is what mine looks like 

Screenshot 2024-05-14 at 2 31 37 PM

File: app/views/students/show.html.erb [hunk 9]
```diff
@@ -1,34 +1,33 @@
...
-            <%= render partial: 'identify', locals: { student: @student } %>
-            <%= render partial: 'engage', locals: { student: @student } %>
-
-          </div>
-        </div>
-        <div class="flex justify-center">
-          <%= render partial: 'strategies', locals: { student: @student } %>
-        </div>
-      </div>
+<div class="py-4 px-2 rounded-lg w-2/5 h-5/6 border border-gray-500 flex flex-col justify-between">
```
Comment (camckin10) [inline app/views/students/show.html.erb:11]:
```suggestion
<div class="py-4 px-2 rounded-lg  h-5/6 border border-gray-500 flex flex-col justify-between">
```
This width here was pushing the content outside of the strategy sidebar.

File: app/views/students/_know.html.erb [hunk 10]
```diff
@@ -1,3 +1,46 @@
...
-</button>
+<div class="grid grid-cols-1 gap-1">
+  <hr>
+  <div class="w-max flex justify-between items-center mb-2">
+    <%= image_tag 'basic_strategy_icons/head-side-gear-duotone 1.png', alt: 'Gear in head icon', class: 'pill', style: 'max-width: 45px; height: auto;' %>
+    <p class="text-white ml-2 text-sm">Ways of Being</p>
+    <button class="strategy-btn bg-gray-300 border-2 border-gray-600 hover:bg-gray-400 text-black font-bold px-2 rounded-full text-sm">
+      Know
+    </button>
+    <button class=" bg-accent-blue-dark border-2 border-gray-600 hover:bg-green-600 text-white font-bold px-2 rounded flex justify-end items-center space-x-1 text-sm">
```
Comment (camckin10) [inline app/views/students/_know.html.erb:9]: You can remove line #9. This is an extra button.

**Comment by camckin10** [comment]
@Zmuse001  Looking good! I left a couple of clean up comments... 
With those changes, your view should look like this: 

Screen Shot 2024-05-15 at 10 50 16 AM

File: app/views/way_of_being/index.html.erb [hunk 11]
```diff
@@ -3,8 +3,12 @@
...
         <ul class="px-4">
             <% @way_of_being.each do |way_of_being| %>
-            <li class="bg-white py-2 border-b border-gray-400 text-gray-400">
-                <%= way_of_being.name %>
+            <li class="flex w-full justify-between bg-white py-2 border-b border-gray-400 text-gray-400">
+                <div><%= way_of_being.name %></div>
+                <%= link_to "#", class:"strategy-btn  border-2 border-gray-600 hover:bg-green-600 text-black px-2 rounded flex items-center space-x-1 text-lg" do %>   
+                <span>Identify</span>
+                <%= show_svg('right_arrow.svg') %>
+                <% end %>
```
Comment (edithemmings) [inline app/views/way_of_being/index.html.erb:11]: ^looks like you combined two tickets on one branch?

Comment (Zmuse001) [reply app/views/way_of_being/index.html.erb:11]: working on it!!

**Comment by Zmuse001** [comment]
i'm going to recreate this PR and clean it up

**Pull request closed without merging**